astropy/units/core.py

```diff
@@ -1755,19 +1760,20 @@ def __call__(self, s, represents=None, format=None, namespace=None,
             if is_effectively_unity(represents.value):
                 represents = represents.unit
             else:
-                # cannot use _error_check=False: scale may be effectively unity
                 represents = CompositeUnit(represents.value *
                                            represents.unit.scale,
                                            bases=represents.unit.bases,
-                                           powers=represents.unit.powers)
+                                           powers=represents.unit.powers,
+                                           _error_check=False)
 
         if isinstance(s, Quantity):
             if is_effectively_unity(s.value):
                 s = s.unit
             else:
                 s = CompositeUnit(s.value * s.unit.scale,
                                   bases=s.unit.bases,
-                                  powers=s.unit.powers)
+                                  powers=s.unit.powers,
+                                  _error_check=False)
```

Comment:
Here it's safe to not check because the unit is coming from a quantity, so it has already been checked?

Reply:
Yes, same logic as above. And similarly, not very important for speed.